onedal/utils/validation.py

```diff
@@ -153,15 +153,6 @@ def _check_array(
 
     if sp.issparse(array):
         return array
-
-    # TODO: Convert this kind of arrays to a table like in daal4py
-    if not array.flags.aligned and not array.flags.writeable:
```

Comment:
From numpy documentation: https://numpy.org/devdocs/dev/alignment.html only structured arrays will cause this, meaning that this is a non issue based on the dtypes of use for oneDAL (float32, float64, int32, and int64), and would failt to convert in table.cpp anyway.

Reply:
I think numpy also allows creating non-aligned arrays out of custom non-owned pointers, for example through [PyArray_SimpleNewFromData](https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_SimpleNewFromData), so in theory there could be a non-aligned float32 array or similar. But it'd be a very unlikely input, since default allocators in most platforms are aligned.

Also an easier conversion could be through [np.require](https://numpy.org/doc/stable/reference/generated/numpy.require.html).

Reply:
Good point, I'll probably because of this too I will make a stricter check on C++ side https://github.com/intel/scikit-learn-intelex/blob/main/onedal/decomposition/pca.py#L150 probably using https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_FromArray where I can specify the requirements to be aligned and owned (rather than use PyArray_GETCONTIGUOUS)

Reply:
With respect to non-ownership, I don't want to step on anyone's toes, as changes to PCA are occuring here which may inpact the ownership check there: https://github.com/intel/scikit-learn-intelex/pull/2106